activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -384,7 +384,30 @@ def insert_fixtures(fixtures, table_name)
         manager.into(table)
         columns.each_key { |column| manager.columns << table[column] }
         manager.values = manager.create_values_list(values)
-        execute manager.to_sql, "Fixtures Insert"
+        manager.to_sql
+      end
+
+
+      # Generate and execute a single SQL string that contains multiple INSERT statement for *all* fixtures.
+      # INSERT INTO table (col1, col2) VALUES (?, ?), (?, DEFAULT);\n
+      # INSERT INTO table2 (col1, col2) VALUES (?, ?), (?, DEFAULT);\n
+      #
+      # Overridden in adapters that require something beyond a simple insert (eg. Oracle).
+      def insert_fixtures(fixture_set, tables_to_delete = [])
+        total_sql = String.new
+
+        fixture_set.each do |table_name, fixtures|
+          total_sql << build_sql_for_fixtures_insert(fixtures, table_name)
+          total_sql << ";\n"
+        end
+
+        disable_referential_integrity do
+          transaction(requires_new: true) do
```

Comment:
Won't this break if there's a surrounding transaction?

I guess the reconnect would break such a transaction no matter what, actually. :confused:

Reply:
> I guess the reconnect would break such a transaction no matter what, actually.

Yes the reconnect would break any open transaction, is this something we should be concerned about?